public/app/features/alerting/unified/components/notification-policies/Policy.tsx

```diff
@@ -162,12 +191,33 @@ const Policy = (props: PolicyComponentProps) => {
   // policies then we should not show it. Same if the user is not supported to see autogenerated policies.
   const hideCurrentPolicy =
     isAutoGenerated && (!isAllowedToSeeAutogeneratedChunk || !isSupportedToSeeAutogeneratedChunk);
+  const hideCurrentPolicyForFilters = filtersApplied && !belongsToMatchPath;
 
-  if (hideCurrentPolicy) {
+  if (hideCurrentPolicy || hideCurrentPolicyForFilters) {
     return null;
   }
+
   const isImmutablePolicy = isDefaultPolicy || isAutogeneratedPolicyRoot;
   // TODO dead branch detection, warnings for all sort of configs that won't work or will never be activated
+
+  const childPoliciesBelongingToMatchPath = childPolicies.filter((child) =>
```

Comment:
What if we split filtering and pagination into two steps? 
First step would be to get all filtered child policies or all when no filters applied
The second step would be only about pagination on the result of the above